docs/extensibility.md

```diff
@@ -0,0 +1,94 @@
+# Extensibility
+
+Extensibility is key for WordPress and like the rest of WordPress components, Gutenberg is hightly extensible.
+
+
+## Creating Blocks
+
+Gutenberg is about blocks and the main extensibility API of Gutenberg is the Block API. It allows you to create static blocks, dynamic blocks rendering on the server and also blocks saving data to Post Meta for more structured content.
+
+If you want to learn more about block creations, The [Blocks Tutorial](./blocks) is the best place to sart.
```

Comment:
I would still add a very simple example for registration to have a better overview on this page.